lib/adapters/tool/playwright/index.ts

```diff
@@ -0,0 +1,311 @@
+        });
+
+        return PlaywrightTestCollectionAdapter.create(tests);
+    }
+
+    async run(_testCollection: PlaywrightTestCollectionAdapter, tests: TestSpec[] = []): Promise<boolean> {
+        return this._runTests(tests);
+    }
+
+    async runWithoutRetries(_testCollection: PlaywrightTestCollectionAdapter, tests: TestSpec[] = []): Promise<boolean> {
+        return this._runTests(tests, ['--retries', '0']);
+    }
+
+    private async _runTests(tests: TestSpec[] = [], runArgs: string[] = []): Promise<boolean> {
+        if (!this._reportBuilder || !this._eventSource) {
+            throw new Error('"reportBuilder" and "eventSource" instances must be initialize before run tests');
+        }
+
+        const queue = new PQueue({concurrency: os.cpus().length});
+
+        return new Promise((resolve, reject) => {
+            const args = ([] as string[]).concat(prepareRunArgs(tests, this._configPath, this._haveProjectsInConfig), runArgs);
+            const child = spawn(this._pwtBinaryPath, args, {
+                stdio: ['inherit', 'inherit', 'inherit', 'ipc']
+            });
+
+            ipc.on<PwtEventMessage>(ClientEvents.BEGIN_STATE, (data) => {
+                data.result.status = TestStatus.RUNNING;
+
+                queue.add(async () => {
+                    const testBranch = await registerTestResult(data, this._reportBuilder);
+
+                    this._eventSource.emit(data.event, testBranch);
+                }).catch(reject);
+            }, child);
+
+            ipc.on<PwtEventMessage>(ClientEvents.TEST_RESULT, (data) => {
+                queue.add(async () => {
+                    const testBranch = await registerTestResult(data, this._reportBuilder);
+
+                    this._eventSource.emit(data.event, testBranch);
+                }).catch(reject);
+            }, child);
+
+            ipc.on(ClientEvents.END, (data) => {
+                queue.onIdle().then(() => this._eventSource.emit(data.event));
+            }, child);
+
+            child.on('error', (data) => reject(data));
+
+            child.on('exit', (code) => {
+                queue.onIdle().then(() => resolve(!code));
+            });
+        });
+    }
+
+    // Can't handle test results here because pwt does not provide api for this, so save instances and use them in custom reporter
+    handleTestResults(reportBuilder: GuiReportBuilder, eventSource: EventSource): void {
+        this._reportBuilder = reportBuilder;
+        this._eventSource = eventSource;
+    }
+
+    updateReference(): void {}
+
+    halt(err: Error): void {
+        logger.error(err);
+        process.exit(1);
+    }
+}
+
+function readPwtConfig(opts: ToolAdapterOptionsFromCli): {configPath: string, config: FullConfig} {
+    const configPaths = opts.configPath ? [opts.configPath] : DEFAULT_CONFIG_PATHS;
+    let originalConfig!: FullConfig;
+    let resolvedConfigPath!: string;
+
+    const revertTransformHook = setupTransformHook();
```

Comment:
Here I need to use pirates with babel in order to correctly transpile config on `.ts`, `.mts` and `.mjs`. Other extensions are skipped